index.js

```diff
@@ -5,9 +5,5 @@ module.exports = function (db, prefix, opts) {
   if (typeof prefix === 'object' && !opts) return module.exports(db, null, prefix)
   if (!opts) opts = {}
 
-  opts.db = function () {
-    return subdown(db, prefix, opts)
-  }
-
-  return levelup(opts)
+  return levelup(subdown(db, prefix, opts), opts)
```

Comment:
@vweevers What if we did `levelup(encoding(subdown(db, prefix, opts)), opts)` here instead?

Reply:
Should work, yeah, with one addition: also pass `opts` to `encoding()`.

Reply:
What happens with `deferred-leveldown` though? Wouldn't every (sub)db be wrapped needlessly?

Reply:
Hmm, not following you. That's already the case?

Reply:
> That's already the case?

It is? OK, then never mind, we can optimize that later if we want

Reply:
Aye, so each `levelup` has its own `deferred-leveldown` and `subleveldown` just returns a new `levelup` so nothing has changed there.

But I agree, it's a bit overhead maybe. Not sure what we can do about it since it's part of `levelup`.

Reply:
Scratch my last comment. I meant using `level-packager` which wraps with `levelup` and `encoding-down`.

Reply:
> This hurts my brain :)

You're not alone :)